File: Java/lab22/src/main/java/org/example/WorkerTemplate.java
package org.example;

public abstract class WorkerTemplate {
    public void startVariable(){
        System.out.println("Worker start varible");
    }
    public void endVarible(){
        System.out.println("Worker finish varible");
    }

    abstract void doWork();
    abstract void takeOver();
    abstract void continueWork();
    public void showWorkSchedule(){
        startVariable();
        doWork();
        takeOver();
        continueWork();
        endVarible();
    }
}

File: Python/lab22/WorkerTemplate.py
from abc import ABC, abstractmethod


class WorkerTemplate(ABC):
    def start_variable(self):
        print("Worker start variable")

    def end_variable(self):
        print("Worker finish variable")

    @abstractmethod
    def do_work(self):
        pass

    @abstractmethod
    def take_over(self):
        pass

    @abstractmethod
    def continue_work(self):
        pass

    def show_work_schedule(self):
        self.start_variable()
        self.do_work()
        self.take_over()
        self.continue_work()
        self.end_variable()
